src/serac/numerics/functional/tensor.hpp

```diff
@@ -756,6 +692,18 @@ SERAC_HOST_DEVICE constexpr auto inner(const tensor<S, m, n>& A, const tensor<T,
   return sum;
 }
 
+template <typename S, typename T, int m>
+SERAC_HOST_DEVICE constexpr auto inner(const tensor<S, m>& A, const tensor<T, m>& B)
+{
+  decltype(S{} * T{}) sum{};
+  for (int i = 0; i < m; i++) {
+    sum += A[i] * B[i];
+  }
+  return sum;
+}
+
+SERAC_HOST_DEVICE constexpr auto inner(double A, double B) { return A * B; }
```

Comment:
same here, I'm not sure these are used any more. Maybe they're still worth keeping

Reply:
I think is still worth keeping as a general purpose capability.

Reply:
Because of the template, I'm unsure of all of the uses this function may have. Is this only for the case of a scalar product of two (arbitrary dimension) vectors? If so, is the motivation for removing it that it does the same the same thing as `dot(...)`? If the answer to this is also "yes", I'm with Jamie, I think we should keep it.